### DISCUSSION TITLE
Rationale behind negative right/left margins for Grid component

### DISCUSSION THREAD

**Discussion by mannehedlund** [twilio-labs/paste#3074] [General] [answered]
Hi Paste folks,

Me and @james2hey noticed that for the Grid component it applies a negative margin equal to the gutter size. As a result, the Grid creates a horizontal scrollbar where you scroll to see the complete padding -- a bit ugly.

(sort of like the same effect as not using `box-sizing: border-box`)

Basically this, which is on the Paste docs page showcase of the component too.
https://paste.twilio.design/components/grid#responsive-column-gutters
![image](https://user-images.githubusercontent.com/46298027/223482143-48d6be6b-fb28-4da3-9e4e-f9cede6526af.png)

We were able to get rid of this problem by adding padding equal to the gutter in a parent div. However, we do wonder if there was some specific reason for adding this negative margin to begin with that we should be aware of.

Here is the Paste code which adds the negative margin.
https://github.com/twilio-labs/paste/blob/046fcc4d1e592f1ce0be733c798831b151037d60/packages/paste-core/layout/grid/src/utils.tsx#L112

Thanks in advance 🙂

**Comment by SiTaggart** [comment, marked as answer]
Hi @mannehedlund 

The main reason if for nesting grids. The new nested grid should still have gutters attached, but the columns should touch the outer wall of the parent column. If you use padding on the grid container you'll indent the child grid.

It's a very popular and common way to create grid systems on the web.

Is it great? No. Better solutions exist in CSS now with CSS grid, which we support on the Box component. We've just not been able to prioritize a CSS Grid version of the Grid Component yet. It's on our wish list, though.

In the meantime, if the negative margins don't work for you, I'd suggest just going with Box and using the CSS grid style props we provide.

**Reply by mannehedlund** [reply]
Thanks, this is valuable context! 😄
